src/components/dialogs/AddStaffNotationDialog.vue

```diff
@@ -215,84 +209,82 @@ export default class AddStaffNotationDialog extends Vue {
+    if (this.isPutBackOn) {
+      filing.business = {
+        identifier: this.getIdentifier,
         legalType: this.getEntityType,
         legalName: this.getEntityName,
         foundingDate: this.getEntityFoundingDate
       }
-      data.filing[this.name] = { ...data.filing[this.name],
+      filing[FilingTypes.PUT_BACK_ON] = {
         details: this.notation
       }
       if (this.courtOrderNumber) {
-        data.filing[this.name] = { ...data.filing[this.name],
-          courtOrder: {
-            effectOfOrder: (this.planOfArrangement ? EffectOfOrderTypes.PLAN_OF_ARRANGEMENT : ''),
-            fileNumber: (this.courtOrderNumber ? this.courtOrderNumber : '')
-          }
+        // add court order sub-object
+        filing[FilingTypes.PUT_BACK_ON].courtOrder = {
+          effectOfOrder: (this.planOfArrangement ? EffectOfOrderTypes.PLAN_OF_ARRANGEMENT : ''),
+          fileNumber: this.courtOrderNumber
         }
       }
-    } else if (this.administrativeDissolution) {
-      data.filing.business = { ...data.filing.business,
+    } else if (this.isAdministrativeDissolution) {
+      filing.business = {
+        identifier: this.getIdentifier,
         legalType: this.getEntityType,
         legalName: this.getEntityName,
         foundingDate: this.getEntityFoundingDate
       }
-      data.filing[this.name] = { ...data.filing[this.name],
-        dissolutionType: 'administrative',
+      filing[FilingTypes.DISSOLUTION] = {
+        dissolutionType: DissolutionTypes.ADMINISTRATIVE,
         dissolutionDate: this.getCurrentDate,
         details: this.notation
       }
       if (this.courtOrderNumber) {
-        data.filing[this.name] = { ...data.filing[this.name],
-          courtOrder: {
-            effectOfOrder: (this.planOfArrangement ? EffectOfOrderTypes.PLAN_OF_ARRANGEMENT : ''),
-            fileNumber: (this.courtOrderNumber ? this.courtOrderNumber : ''),
-            orderDetails: this.notation
-          }
+        // add court order sub-object
+        filing[FilingTypes.DISSOLUTION].courtOrder = {
+          effectOfOrder: (this.planOfArrangement ? EffectOfOrderTypes.PLAN_OF_ARRANGEMENT : ''),
+          fileNumber: this.courtOrderNumber,
+          orderDetails: this.notation
         }
       }
     } else {
-      data.filing[this.name] = { ...data.filing[this.name],
+      // this may be a Registrar's Notation, a Registrar's Order, or a Court Order filing
+      filing.business = {
+        identifier: this.getIdentifier
+      }
+      // NB: no need to nest this as this filing _is_ a court order object
+      filing[this.name] = {
         effectOfOrder: (this.planOfArrangement ? EffectOfOrderTypes.PLAN_OF_ARRANGEMENT : ''),
         fileNumber: (this.courtOrderNumber ? this.courtOrderNumber : ''),
         orderDetails: this.notation
       }
     }
 
-    const url = `businesses/${this.getIdentifier}/filings`
     let success = false
-    await axios.post(url, data).then(() => {
```

Comment:
All API calls should be centralized in a service file. Many components in this project make network calls directly. I am fixing just one component at a time :)